### PR TITLE
WIP: support building and packaging on Windows with VS2015

### PR DESCRIPTION
**Pull request by alex85k**
Have tried to adapt setup.py for Windows (Python 3).
Todo:
- [ ] Check and review code changes for other systems (fix clang)
- [x] Add Appveyor build
- [x] Fix one failing test

### CODE REVIEW

**Comment by alex85k** [comment]
So, I somehow broke the clang compilation (finding boost) and appveyor builds still gives the strange linking error. 

However, on home pc with Python 3.6.1 x64 the same development pack is successful: https://github.com/alex85k/pyosmium_libs/releases , the batch file is here 
https://github.com/alex85k/winbuilds/blob/master/build_pyosmium.bat .

**Comment by alex85k** [comment]
Have located the problem on appveyor: this is VS2015 Update 3 bug:
https://connect.microsoft.com/VisualStudio/Feedback/Details/2852624

Trying to switch to 2017.

File: setup.py [hunk 1]
```diff
@@ -54,13 +75,15 @@ def get_versions():
...
-    lib = find_library("boost_python%s" % suf)
-    if lib is not None:
-        libs.append("boost_python%s" % suf)
+    for suf in suffixes:
+        print("finding boost_python%s" % suf )
+        lib = find_library("boost_python%s" % suf)
+        print(lib)
+        if lib is not None:
+           libs.append("boost_python%s" % suf)
         break
```
Comment (lonvia) [inline setup.py]: You forgot  a level of indent here for the break. That's what breaks the Mac build.

Comment (alex85k) [inline setup.py]: Oops, thank you very much!

File: setup.py [hunk 2]
```diff
@@ -54,14 +75,15 @@ def get_versions():
...
-    raise Exception("Cannot find boost_python library")
-
+    for suf in suffixes:
+        print("finding boost_python%s" % suf )
+        lib = find_library("boost_python%s" % suf)
+        print(lib)
+        if lib is not None:
+           libs.append("boost_python%s" % suf)
+           break
+        else:
```
Comment (lonvia) [inline setup.py]: Sorry, I missed that that's wrong, too. That `else:` branch actually belongs to the for loop (i.e. remove one level of indention).

**Comment by alex85k** [comment]
Hm, that "else" condition for loops in Python was a surprise :) Tried to fix again.

**Comment by alex85k** [comment]
The bug workaround seemed to work on Appveyor:
https://ci.appveyor.com/project/alex85k/pyosmium/build/1.0.13
now trying to fix tests...

**Comment by alex85k** [comment]
Appveyor is almost happy:
https://ci.appveyor.com/project/alex85k/pyosmium/build/1.0.14

There is only one failing test 
```
ERROR: test_func (test_osm.TestAreaFromWayAttributes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\dev\pyosmium\test\helpers.py", line 118, in test_func
    self.Handler().apply_file(fn, self.apply_locations, self.apply_idx)
  File "C:\dev\pyosmium\test\test_osm.py", line 129, in area
    assert_equals(n.num_rings(), (1, 0))
TypeError: No to_python (by-value) converter found for C++ type: struct std::pair<unsigned __int64,unsigned __
```

**Comment by alex85k** [comment]
Appveyor is finally happy, test passing!
https://ci.appveyor.com/project/alex85k/pyosmium/build/1.0.16
However, this means that long is 32 bit , and somewhere on Windows __int64 is somehow used. Hope it works correctly for big IDs etc.

**Comment by lonvia** [comment]
Many thanks. `setup.py` still needs some cleanup. I'd be happy to do the finishing up. Just need to wait that we have appveyor enabled on this repo.

I'm not sure why you say 'long is 32bit'. Where does that come from, python, compiler, boost-python? I can add a test for big IDs.

**Comment by alex85k** [comment]
Yes, I'd be happy if you can clean up setup.py. That was the first Python module :)

long is certainly 32 bit on VS 64 bit, elsewhere 
```
std_pair_to_python_converter<unsigned long, unsigned long>();
std_pair_to_python_converter<uint64_t, uint64_t>();
```
was the same.

**Comment by lonvia** [comment]
Ah, so this is just about checking the code for correct use of 'long' in the code. The code should be following the conventions from libosmium there, so we are hopefully fine. In any case, I've pushed two tests with 64bit ids now.

**Comment by alex85k** [comment]
I have tried rebasing to current master, all additional tests pass successfuly. This time tested on Python 2.7.

**Comment by alex85k** [comment]
Pushed the reabsed version here to trigger Appveyor.

**Comment by alex85k** [comment]
Cleaned the PR after https://github.com/osmcode/pyosmium/pull/36, only thing to fix after is supporting Python 2.7.
Boost Python and Osmium certainly can not be built with 2.7-native Visual C++ 9.0.
However, substitution of VS2015 works for this extensions without problems. The only problem on Appveor is auto-linking.

**Comment by alex85k** [comment]
DEFAULTLIB always wants to use boost_python-vc140-mt-1_63.lib and by default boost python for Python3 uses boost_python3-vc140-mt-1_63.lib .

Should we just use different development packs or change the library name?

**Comment by alex85k** [comment]
Re-pushed. Seems to work with two different sets of libraries. Linker debug output removed.

**Comment by alex85k** [comment]
You can also fork https://github.com/alex85k/pyosmium_libs to create devpacks on appveyor (or switch to windows-builds) and do not rely on my github release binaries.

**Comment by lonvia** [comment]
> Should we just use different development packs or change the library name?

The different packages are better. That should also allow us to add more python versions in the future. So I've merged as is. Thanks again.

> You can also fork https://github.com/alex85k/pyosmium_libs to create devpacks on appveyor

It would be more practical to have that under the osmcode org. You can also just move the repo here to osmcode (keeping full access rights, of course). Or I can just fork it, as you prefer.

**Pull request merged**